cmake.md

```diff
@@ -0,0 +1,63 @@
+## CMake build system
+### Terminology
+
+The CMake system in symbiflow-arch-defs uses the following hierarchy:
+
+* `ARCH` - Architecture is the high level grouping of a set of "similar" FPGA
+  types. An example would be the "Lattice iCE40 Series" (ice40) or the "Xilinx Artix 7 Series" (artix7).
+* `DEVICE_TYPE` - A type of device.  This is defined as an `ARCH` and the
+  arch.xml which defines the units with this device type.
+* `DEVICE` - A specific instance of a `DEVICE_TYPE`.  A device definition is
+   a `DEVICE_TYPE`, a `PACKAGE` list, and IO map definitions for each package
+   if needed.  Each `DEVICE` should coorispond with a layout definition within
+   the `DEVICE_TYPE`.
+* `BOARD` - A specific `DEVICE` and `PACKAGE` instance, along with a
+  `PROG_TOOL`/`PROG_CMD` (e.g. commands to program a specified board).
+
+### Creating a new FPGA target
```

Comment:
`### Adding new tests`?

Reply:
ADD_FPGA_TARGET isn't really just about tests, it also the way designs are compiled.  With a small amount of refactoring, this outline should work for a frozen repository of arch.merge.xml, rr_graph files and IO pinmaps.